pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,517 @@
+			return err
+		}
+		if err := r.client.Create(context.TODO(), newSS); err != nil {
+			logger.Error(err, "Error creating syncset")
+			return err
+		}
+		logger.Info("Done creating a new SyncSet")
+
+	} else {
+		logger.Info("SyncSet Already Present, nothing to do here...")
+		// return directly if the syscset already existed
+
+	}
+	return nil
+
+}
+
+func newDMSSecret(namespace string, name string, snitchURL string) *corev1.Secret {
+
+	dmsSecret := &corev1.Secret{
+		Type: "Opaque",
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			config.KeySnitchURL: []byte(snitchURL),
+		},
+	}
+
+	return dmsSecret
+
+}
+
+func newSyncSet(namespace string, dmsSecret string, clusterDeploymentName string, dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration) *hivev1.SyncSet {
+
+	newSS := &hivev1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dmsSecret,
+			Namespace: namespace,
+		},
+		Spec: hivev1.SyncSetSpec{
+			ClusterDeploymentRefs: []corev1.LocalObjectReference{
+				{
+					Name: clusterDeploymentName,
+				},
+			},
+			SyncSetCommonSpec: hivev1.SyncSetCommonSpec{
+				ResourceApplyMode: hivev1.SyncResourceApplyMode,
+				Secrets: []hivev1.SecretMapping{
+					{
+						SourceRef: hivev1.SecretReference{
+							Name:      dmsSecret,
+							Namespace: namespace,
+						},
+						TargetRef: hivev1.SecretReference{
+							Name:      dmsi.Spec.TargetSecretRef.Name,
+							Namespace: dmsi.Spec.TargetSecretRef.Namespace,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return newSS
+
+}
+
+// delete snitches,secrets and syncset associated with the dmsi that has been deleted
+func (r *ReconcileDeadmansSnitchIntegration) deleteDMSI(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clustDeploy *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", clustDeploy.Name, "cluster-deployment.Namespace:", clustDeploy.Namespace)
+	deadMansSnitchFinalizer := "dms.managed.openshift.io/deadmanssnitch-" + dmsi.Name
+
+	// Delete the dms
+	logger.Info("Deleting the DMS from api.deadmanssnitch.com")
+	snitchName := clustDeploy.Spec.ClusterName + "." + clustDeploy.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
+	snitches, err := dmsc.FindSnitchesByName(snitchName)
+	if err != nil {
+		return err
+	}
+	for _, s := range snitches {
+		delStatus, err := dmsc.Delete(s.Token)
+		if !delStatus || err != nil {
+			logger.Info("Failed to delete the DMS from api.deadmanssnitch.com")
+			return err
+		}
+		logger.Info("Deleted the DMS from api.deadmanssnitch.com")
+	}
+
+	// Delete the SyncSet
+	logger.Info("Deleting DMS SyncSet")
+	err = utils.DeleteSyncSet(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
```

Comment:
```suggestion
	err = utils.DeleteSyncSet(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+"-"+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
```

You've removed the hyphen from the beginning of `config.RefSecretPostfix`, so it should be added here.

Reply:
Good catch thanks